Please find below the Project for Statistical Learning course. This is an individual assignment. Kindly submit it before it's 
deadline.

The Titan Insurance Company has just installed a new incentive payment scheme for its lift policy sales force. It wants to
have an early view of the success or failure of the new scheme. Indications are that the sales force is selling more policies, 
but sales always vary in an unpredictable pattern from month to month and it is not clear that the scheme has made a 
significant difference.

Life Insurance companies typically measure the monthly output of a salesperson as the total sum assured for the policies 
sold by that person during the month. For example, suppose salesperson X has, in the month, sold seven policies for which
the sums assured are £1000, £2500, £3000, £5000, £10000, £35000. X's output for the month is the total of these sums assured,
£61,500. Titan's new scheme is that the sales force receives low regular salaries but are paid large bonuses related to their
output (i.e. to the total sum assured of policies sold by them). The scheme is expensive for the company, but they are looking 
for sales increases which more than compensate. The agreement with the sales force is that if the scheme does not at least
break even for the company, it will be abandoned after six months.

The scheme has now been in operation for four months. It has settled down after fluctuations in the first two months due to
the changeover.

To test the effectiveness of the scheme, Titan have taken a random sample of 30 salespeople measured their output in the 
penultimate month prior to changeover and then measured it in the fourth month after the changeover (they have deliberately 
chosen months not too close to the changeover). The outputs of the salespeople are shown in Table 1

SALESPERSON  Old Scheme (in thousands) New Scheme (in thousands)



Questions 
Find the mean of old scheme and new scheme column. (5 points)
Use the five percent significance test over the data to determine the p value to check new scheme has significantly raised
outputs? (10 points)
What conclusion does the test (p-value) lead to? (2.5 points)
Suppose it has been calculated that in order for Titan to break even, the average output must increase by £5000 in the scheme 
compared to the old scheme. If this figure is alternative hypothesis, what is:
        a) The probability of a type 1 error? (2.5 points)

        b) What is the p- value of the hypothesis test if we test for a difference of $5000? (10 points)

        c) Power of the test (5 points)

In [1]:
import pandas as pd
import numpy as np
import scipy.stats as stats

In [55]:
scheme = pd.read_csv('data.csv')
#Since given data are in 000, it will be better to convert them in thousands.
scheme=scheme*1000
scheme

SALESPERSON  Old Scheme (in thousands)  New Scheme (in thousands)
0          1000                      57000                      62000
1          2000                     103000                     122000
2          3000                      59000                      54000
3          4000                      75000                      82000
4          5000                      84000                      84000
5          6000                      73000                      86000
6          7000                      35000                      32000
7          8000                     110000                     104000
8          9000                      44000                      38000
9         10000                      82000                     107000
10        11000                      67000                      84000
11        12000                      64000                      85000
12        13000                      78000                      99000
13        14000                      53000                      39000
14        15000                      41000                      34000
15        16000                      39000                      58000
16        17000                      80000                      73000
17        18000                      87000                      53000
18        19000                      73000                      66000
19        20000                      65000                      78000
20        21000                      28000                      41000
21        22000                      62000                      71000
22        23000                      49000                      38000
23        24000                      84000                      95000
24        25000                      63000                      81000
25        26000                      77000                      58000
26        27000                      67000                      75000
27        28000                     101000                      94000
28        29000                      91000                     100000
29        30000                      50000                      68000

In [48]:
new=np.array(scheme['New Scheme (in thousands)'])
new

array([ 62000, 122000,  54000,  82000,  84000,  86000,  32000, 104000,
        38000, 107000,  84000,  85000,  99000,  39000,  34000,  58000,
        73000,  53000,  66000,  78000,  41000,  71000,  38000,  95000,
        81000,  58000,  75000,  94000, 100000,  68000], dtype=int64)

In [49]:
old=np.array(scheme['Old Scheme (in thousands)'])
old

array([ 57000, 103000,  59000,  75000,  84000,  73000,  35000, 110000,
        44000,  82000,  67000,  64000,  78000,  53000,  41000,  39000,
        80000,  87000,  73000,  65000,  28000,  62000,  49000,  84000,
        63000,  77000,  67000, 101000,  91000,  50000], dtype=int64)

In [50]:
diff=new-old
diff

array([  5000,  19000,  -5000,   7000,      0,  13000,  -3000,  -6000,
        -6000,  25000,  17000,  21000,  21000, -14000,  -7000,  19000,
        -7000, -34000,  -7000,  13000,  13000,   9000, -11000,  11000,
        18000, -19000,   8000,  -7000,   9000,  18000], dtype=int64)

In [83]:
diff_mean=diff.mean()
diff_mean

4000.0

In [81]:
s=np.std(diff,ddof = 1)
s

14081.046199376151

In [76]:
se=s/np.sqrt(n)
se

2570.8355455569017

In [53]:
scheme.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Data columns (total 3 columns):
SALESPERSON                  30 non-null int64
Old Scheme (in thousands)    30 non-null int64
New Scheme (in thousands)    30 non-null int64
dtypes: int64(3)
memory usage: 800.0 bytes


In [77]:
xbar=diff.mean()
xbar

4000.0

In [78]:
print("T-Statistics")
print((xbar-mu)/se)

T-Statistics
1.4999014645896778


In [80]:
print("T-critical Values")
print(stats.t.isf(0.025,loc=0,scale=se,df=n-1))
print(stats.t.isf(0.975,loc=0,scale=se,df=n-1))

T-critical Values
5257.949062821375
-5257.949062821374


In [ ]:
T-statistics and xbar lies within critical region, so we fail to reject null.

In [54]:
scheme.head(5)

SALESPERSON  Old Scheme (in thousands)  New Scheme (in thousands)
0         1000                      57000                      62000
1         2000                     103000                     122000
2         3000                      59000                      54000
3         4000                      75000                      82000
4         5000                      84000                      84000

# Find the mean of old scheme and new scheme column.

In [56]:
old_scheme_mean=scheme['Old Scheme (in thousands)'].mean()
old_scheme_mean

68033.33333333333

In [57]:
new_scheme_mean=scheme['New Scheme (in thousands)'].mean()
new_scheme_mean

72033.33333333333

In [58]:
std_new=scheme['New Scheme (in thousands)'].std()
std_new

24062.39494677769

In [59]:
std_old=scheme['Old Scheme (in thousands)'].std()
std_old

20455.98021207445

In [60]:
import scipy.stats as stats

In [73]:
n=30
alpha=0.05

In [62]:
df=n-1
df

29

In [63]:
std1=std_new - std_old
std1

3606.4147347032413

In [64]:
Zcrit = round(stats.norm.isf(q = 0.05),2)
print('Value of Z critical is %3.6f' %Zcrit)

Value of Z critical is 1.640000


# Use the five percent significance test over the data to determine the p value to check new scheme has significantly raised outputs?

In [65]:
from    scipy.stats             import  ttest_1samp,ttest_ind, wilcoxon, ttest_ind_from_stats
import  scipy.stats             as      stats  
from    statsmodels.stats.power import  ttest_power
import  matplotlib.pyplot       as      plt

In [72]:
t_statistic, p_value  =  stats.ttest_rel(new,old)
print(t_statistic)
print(p_value) 

1.5559143823544377
0.13057553961337662


In [74]:
import  scipy.stats  as stats  
t_statistic, p_value  =  stats.ttest_1samp(diff, 0)
print(t_statistic)
print(p_value) 

1.5559143823544377
0.13057553961337662


In [84]:
print("P-values")
print((1-stats.t.cdf(diff_mean,loc=0,scale=se,df=n-1))*2)

P-values
0.13057553961337676


# What conclusion does the test (p-value) lead to?

In this example, p value is 0.130 and it is more than 5% level of significance
So the statistical decision is to accept the null hypothesis at 5% level of significance.
Hence accept the null hypothesis that the new scheme did not
raise the output significantly.

In [85]:
Zcrit

1.64

In [86]:
mu=0
Tstatistics=1.499

In [96]:
#Tstat=(((d-mu)/s)/np.sqrt(n))
d1=s/np.sqrt(n)
d=1.499*d1
d


3853.682482789796

In [102]:
#Probability of rejecting the null hypothesis when it is
#true. The probability of a Type I error in hypothesis testing is predetermined
#by the significance level.
#Type 1 error is 13.05%

# Power of the test 

Power of the hypothesis test is 1 minus the probability of Type II error.
Basically the power of a test is the probability that we make the right decision
when the null is not correct (i.e. we correctly reject it)

In [98]:
Power = 1- 0.13057
Power

0.86943

Thus 86.94 % is the power of test

# What is the p- value of the hypothesis test if we test for a difference of $5000?

In [ ]:
Now, considering 5000k

In [99]:
Tstat=(d-5000)/d1
Tstat

-0.4458929779430468

In [100]:
print("P-values")
print((1-stats.t.cdf(diff_mean,loc=5000,scale=se,df=n-1))*2)

P-values
1.2998665087386714
